connectors/filtering/basic_rule.py

```diff
@@ -147,6 +147,50 @@ def from_string(cls, string):
                 )
 
 
+class BasicRuleValidationResult:
+    def __init__(self, rule_id, is_valid, validation_message):
+        self.rule_id = rule_id
+        self.is_valid = is_valid
+        self.validation_message = validation_message
+
+    @classmethod
+    def valid_result(cls, rule_id):
+        return BasicRuleValidationResult(
+            rule_id=rule_id, is_valid=True, validation_message="Valid rule"
+        )
+
+
+class BasicRuleValidator:
+    @classmethod
+    def validate(cls, rule):
+        raise NotImplementedError
+
+
+class BasicRuleNoMatchAllRegexValidator(BasicRuleValidator):
+    MATCH_ALL_REGEXPS = [".*", "(.*)"]
+
+    @classmethod
+    def validate(cls, basic_rule_json):
+        basic_rule = BasicRule.from_json(basic_rule_json)
```

Comment:
Should the `basic_rule_json` be validated before calling `.from_json`?

Reply:
No, as we'll always execute the JSON against the schema validation first. Therefore we can reuse functionality like `is_default_rule` and compare against enums without having to implement these again in this method (which would duplicate functionality we already have in `BasicRule`)

Reply:
No as `BasicRuleNoMatchAllRegexValidator` doesn't use a schema to validate against. The class performs a validation checking dependencies between fields dependant one the value of one field (this is not possible with fastjsonschema, you can only specify dependencies in the sense of: field X can only exist if field Y exists, without looking at the value). Therefore this validation should be separated from the schema validation.